src/core/loader/cia.cpp

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <cstring>
+#include <memory>
+#include <string>
+#include "common/common_types.h"
+#include "common/file_util.h"
+#include "common/logging/log.h"
+#include "common/symbols.h"
+#include "core/hle/kernel/process.h"
+#include "core/hle/kernel/resource_limit.h"
+#include "core/loader/cia.h"
+#include "core/memory.h"
+#include "cia.h"
+namespace Loader {
+/*
+ * This is the current version of the CIA format, it was finalised in late 2010. (Older versions of the CIA format can be viewed on the Talk page)
+
+*The CIA format has a similar structure to the WAD format.
+
+*The file is represented in little-endian.
+
+*The data is aligned in 64 byte blocks (if a content ends at the middle of the block, the next content will begin from a new block).
+ * CIA Header[edit]
+*START SIZE    DESCRIPTION
+*0x00 0x04 Archive Header Size (Usually = 0x2020 bytes)
+*0x04 0x02 Type
+*0x06 0x02 Version
+*0x08 0x04 Certificate chain size
+*0x0C 0x04 Ticket size
+*0x10 0x04 TMD file size
+*0x14 0x04 Meta size (0 if no Meta data is present)
+*0x18 0x08 Content size
+*0x20 0x2000 Content Index
+*The order of the sections in the CIA file:
```

Comment:
going to implement this once i get the format down so that i dont need to keep switching windows